app/models/site.rb

```diff
@@ -85,17 +86,20 @@ def create_template_content
   def create_context
     return nil unless self.contexts.empty?
 
-    #context = Context.new({name: self.name})
-    #context.sites << self
-    #context.save!
-
-
     context = Context.new({name: self.name})
     context.save!(validate: false)
     #context = Context.create!({name: self.name})
     self.context_sites.create(context_id: context.id, is_site_default_context: true)
   end
 
+  def handle_non_compliant_slugs
+    return unless self.slug.blank?
+
+    write_attribute(:slug, self.id)
+
+    self.save!(validate: false)
```

Comment:
if we could handle this as a before hook we wouldn't need this extra database call; if we need to do it as an after hook I'd use `update_attribute` (which skips validations) rather than `write_attribute` + `save`